### PR TITLE
Ensure that EXFLAG_INVALID_POLICY is checked even in leaf certs 

### PR DESCRIPTION
**Pull request by mattcaswell**
Even though we check the leaf cert to confirm it is valid, we
later ignored the invalid flag and did not notice that the leaf
cert was bad.

Fixes: CVE-2023-0465

### CODE REVIEW

File: crypto/x509/x509_vfy.c [hunk 1]
```diff
@@ -1673,13 +1673,15 @@ static int check_policy(X509_STORE_CTX *ctx)
...
+        for (i = 0; i < sk_X509_num(ctx->chain); i++) {
             X509 *x = sk_X509_value(ctx->chain, i);
 
             CB_FAIL_IF((x->ex_flags & EXFLAG_INVALID_POLICY) != 0,
                        ctx, x, i, X509_V_ERR_INVALID_POLICY_EXTENSION);
         }
-        return 1;
+        /* Should not be able to get here */
+        ERR_raise(ERR_LIB_X509, ERR_R_INTERNAL_ERROR);
+        return 0;
```
Comment (t8m) [inline crypto/x509/x509_vfy.c:1684]: I am not sure this is correct. The callback can ignore the error and in such case this should not return 0. Shouldn't it?

**Review by t8m — CHANGES_REQUESTED** [review]

File: crypto/x509/x509_vfy.c (hunk 1 above)
Comment (t8m) [reply crypto/x509/x509_vfy.c:1684]: Maybe add some check that we called the callback at least once in this case - this is the _shouldn't happen_ situation.

Comment (mattcaswell) [reply crypto/x509/x509_vfy.c:1684]: Ah....looks like you're right. I'll create a fixup...

Comment (mattcaswell) [reply crypto/x509/x509_vfy.c:1684]: Fixed

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you.

**Pull request closed without merging**